ci/miri.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -e
+
+MIRI_NIGHTLY=nightly-$(curl -s https://rust-lang.github.io/rustup-components-history/x86_64-unknown-linux-gnu/miri)
+echo "Installing latest nightly with Miri: $MIRI_NIGHTLY"
+rustup set profile minimal
+rustup default "$MIRI_NIGHTLY"
+rustup component add miri
+
+cargo miri test
```

Comment:
Since the other comment mentions big-endian... I could also make Miri do a big-endian test of this by passing `--target mips64-unknown-linux-gnuabi64`.

Reply:
A big-endian test wouldn't hurt.

Reply:
Okay, done. :)